[BD-13][BB-6734] refactor: Remove prefer_xmodules()

0x29a opened the pull request:
## Description

* Removes `prefer_xmodules`. This function is no longer needed as all XModules have been converted to XBlocks.
* Removes `dump_xml_courses` management command and it's parent directory.

`XBLOCK_SELECT_FUNCTION` is removed too, as it could take only `prefer_xmodules` or `default_select` values.

## Supporting information

- Sandbox URL: https://extpr31112.sandbox.opencraft.hosting/ (`staff@example.com;edx`).

## Testing instructions

Create a course, add a few XBlocks, test that import and export work.

openedx-webhooks commented:
Thanks for the pull request, @0x29a!


When this pull request is ready, tag your edX technical lead.


<!-- comment:welcome-blended -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

0x29a committed:
refactor: remove prefer_xmodules

This function is no longer needed as all XModules have been converted to XBlocks.

XBLOCK_SELECT_FUNCTION Django setting is removed too, as it could take only `prefer_xmodules` or `default_select` values.

Agrendalath commented on xmodule/modulestore/__init__.py:
```diff
@@ -1393,12 +1393,3 @@ def only_xmodules(identifier, entry_points):
     from_xmodule = [entry_point for entry_point in entry_points if entry_point.dist.key == 'xmodule']
 
     return default_select(identifier, from_xmodule)
```
@0x29a, just thinking - `only_xmodules` seems to be used only in tests, so we could move it somewhere else (e.g. `xmodules.tests.helpers`). We could also remove it if it's not needed anymore. What do you think?

If it makes sense, we can create a small follow-up for this. We don't need to extend the scope of the current ticket.

Agrendalath commented on lms/djangoapps/debug/management/commands/dump_xml_courses.py:
```diff
@@ -31,7 +31,6 @@ def handle(self, *args, **options):
             default_class='xmodule.hidden_module.HiddenDescriptor',
             load_error_modules=True,
             xblock_mixins=settings.XBLOCK_MIXINS,
-            xblock_select=settings.XBLOCK_SELECT_FUNCTION,
```
Side note: I wasn't able to run this management command, as it doesn't seem to be working without `argparse`. Using `BaseCommand.args` [was deprecated in Django 1.8](https://docs.djangoproject.com/en/1.8/howto/custom-management-commands/#django.core.management.BaseCommand.args) (I didn't find its removal in the release notes, though).

ormsbee replied:
Optional: Please feel free to just delete this entire file. The XMLModuleStore was killed off as a functional thing sometime in late 2015. Nothing about this command should work.

Agrendalath reviewed (approved):
:+1:

* [x] I tested this: checked that XBlocks are rendered/imported/exported correctly
* [x] I read through the code
* [x] I checked for accessibility issues: n/a
* [x] Includes documentation: n/a
* [x] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository: n/a

Agrendalath commented:
@ormsbee, this is ready for your review.

ormsbee reviewed (approved):
Nit on the commit message: You might want to mention that `XBLOCK_SELECT_FUNCTION` is a setting.

Please feel free to get rid of the `dump_xml_courses` management command. It's not required that you do so (I don't know if that breaks other things), but it hasn't been a useful thing for the last seven years.

0x29a committed:
refactor: remove lms/djangoapps/debug/management directory

It contained only one management command, "dump_xml_courses", which
didn't work since the late 2015.

0x29a commented:
@Agrendalath and @ormsbee, thank you for the review. I've modified the commit message and removed the management command.

openedx-webhooks commented:
@0x29a 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/31112&entry.1484655318=https://github.com/0x29a&entry.752974735=2022-10-05+18:34&entry.1917517419=2022-11-16+17:34&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
